SuppressWarningsHolder fails on @SuppressWarnings in annotation array initialization

Check documentation: https://checkstyle.sourceforge.io/config_annotation.html#SuppressWarningsHolder

Identified at https://github.com/checkstyle/checkstyle/pull/9229#issuecomment-775176345:

```bash
➜  checkstyle git:(issue-9229) ✗ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="SuppressWarningsHolder"/>
    </module>
</module>

➜  checkstyle git:(issue-9229) ✗ cat TrailingComma.java 
import java.lang.annotation.*;

@interface TestAnnotation {
    SuppressWarnings[] value() default {@SuppressWarnings({"",})};
}

public class TrailingComma {
}

➜  checkstyle git:(issue-9229) ✗ javac TrailingComma.java

➜  checkstyle git:(issue-9229) ✗ java -jar target/checkstyle-8.41-SNAPSHOT-all.jar -c config.xml TrailingComma.java 
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing TrailingComma.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:311)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:221)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:408)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:331)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:190)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:125)
Caused by: java.lang.IllegalArgumentException: Unexpected container AST: {[4x39]
        at com.puppycrawl.tools.checkstyle.checks.SuppressWarningsHolder.getAnnotationTarget(SuppressWarningsHolder.java:422)
        at com.puppycrawl.tools.checkstyle.checks.SuppressWarningsHolder.visitToken(SuppressWarningsHolder.java:312)
        at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:340)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:451)
        at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:278)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:151)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:87)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:337)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:298)
        ... 5 more
Checkstyle ends with 1 errors.


```

Filter in `openjdk14-excluded.files` for `TrailingComma.java` should be removed while fixing this issue:

```xml
<!-- until https://github.com/checkstyle/checkstyle/issues/9260 -->
<module name="BeforeExecutionExclusionFileFilter">
  <property name="fileNamePattern" value="annotations[\\/]pos[\\/]TrailingComma.java$"/>
</module>

```

--- a/.ci/openjdk14-excluded.files
+++ b/.ci/openjdk14-excluded.files
@@ -58,7 +58,7 @@
 
 <!-- until https://github.com/checkstyle/checkstyle/issues/8653 -->
 <module name="BeforeExecutionExclusionFileFilter">
-  <property name="fileNamePattern" value="[\\/]TrailingComma.java$"/>
+  <property name="fileNamePattern" value="enum[\\/]TrailingComma.java$"/>
 </module>
 
 <!-- until https://github.com/checkstyle/checkstyle/issues/8943 -->

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java
@@ -25,6 +25,7 @@ import java.util.LinkedList;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Optional;
 import java.util.regex.Pattern;
 
 import com.puppycrawl.tools.checkstyle.StatelessCheck;
@@ -307,35 +308,42 @@ public class SuppressWarningsHolder
             identifier = identifier.substring(JAVA_LANG_PREFIX.length());
         }
         if ("SuppressWarnings".equals(identifier)) {
-            final List<String> values = getAllAnnotationValues(ast);
-            if (!isAnnotationEmpty(values)) {
-                final DetailAST targetAST = getAnnotationTarget(ast);
+            getAnnotationTarget(ast).ifPresent(targetAST -> {
+                addSuppressions(getAllAnnotationValues(ast), targetAST);
+            });
+        }
+    }
 
-                // get text range of target
-                final int firstLine = targetAST.getLineNo();
-                final int firstColumn = targetAST.getColumnNo();
-                final DetailAST nextAST = targetAST.getNextSibling();
-                final int lastLine;
-                final int lastColumn;
-                if (nextAST == null) {
-                    lastLine = Integer.MAX_VALUE;
-                    lastColumn = Integer.MAX_VALUE;
-                }
-                else {
-                    lastLine = nextAST.getLineNo();
-                    lastColumn = nextAST.getColumnNo() - 1;
-                }
+    /**
+     * Method to populate list of suppression entries.
+     *
+     * @param values
+     *            - list of check names
+     * @param targetAST
+     *            - annotation target
+     */
+    private static void addSuppressions(List<String> values, DetailAST targetAST) {
+        // get text range of target
+        final int firstLine = targetAST.getLineNo();
+        final int firstColumn = targetAST.getColumnNo();
+        final DetailAST nextAST = targetAST.getNextSibling();
+        final int lastLine;
+        final int lastColumn;
+        if (nextAST == null) {
+            lastLine = Integer.MAX_VALUE;
+            lastColumn = Integer.MAX_VALUE;
+        }
+        else {
+            lastLine = nextAST.getLineNo();
+            lastColumn = nextAST.getColumnNo() - 1;
+        }
 
-                // add suppression entries for listed checks
-                final List<Entry> entries = ENTRIES.get();
-                for (String value : values) {
-                    String checkName = value;
-                    // strip off the checkstyle-only prefix if present
-                    checkName = removeCheckstylePrefixIfExists(checkName);
-                    entries.add(new Entry(checkName, firstLine, firstColumn,
-                            lastLine, lastColumn));
-                }
-            }
+        final List<Entry> entries = ENTRIES.get();
+        for (String value : values) {
+            // strip off the checkstyle-only prefix if present
+            final String checkName = removeCheckstylePrefixIfExists(value);
+            entries.add(new Entry(checkName, firstLine, firstColumn,
+                    lastLine, lastColumn));
         }
     }
 
@@ -362,7 +370,7 @@ public class SuppressWarningsHolder
      */
     private static List<String> getAllAnnotationValues(DetailAST ast) {
         // get values of annotation
-        List<String> values = null;
+        List<String> values = Collections.emptyList();
         final DetailAST lparenAST = ast.findFirstToken(TokenTypes.LPAREN);
         if (lparenAST != null) {
             final DetailAST nextAST = lparenAST.getNextSibling();
@@ -392,36 +400,32 @@ public class SuppressWarningsHolder
     }
 
     /**
-     * Checks that annotation is empty.
-     *
-     * @param values list of values in the annotation
-     * @return whether annotation is empty or contains some values
-     */
-    private static boolean isAnnotationEmpty(List<String> values) {
-        return values == null;
-    }
-
-    /**
      * Get target of annotation.
      *
      * @param ast the AST node to get the child of
      * @return get target of annotation
      */
-    private static DetailAST getAnnotationTarget(DetailAST ast) {
-        final DetailAST targetAST;
+    private static Optional<DetailAST> getAnnotationTarget(DetailAST ast) {
+        final Optional<DetailAST> result;
         final DetailAST parentAST = ast.getParent();
         switch (parentAST.getType()) {
             case TokenTypes.MODIFIERS:
             case TokenTypes.ANNOTATIONS:
             case TokenTypes.ANNOTATION:
             case TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR:
-                targetAST = parentAST.getParent();
+                result = Optional.of(parentAST.getParent());
+                break;
+            case TokenTypes.LITERAL_DEFAULT:
+                result = Optional.empty();
+                break;
+            case TokenTypes.ANNOTATION_ARRAY_INIT:
+                result = getAnnotationTarget(parentAST);
                 break;
             default:
                 // unexpected container type
                 throw new IllegalArgumentException("Unexpected container AST: " + parentAST);
         }
-        return targetAST;
+        return result;
     }
 
     /**

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder7.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder7.java
@@ -8,21 +8,27 @@ public class InputSuppressWarningsHolder7 {
     @TestSwAnnotation(@SuppressWarnings("unchecked"))
     private List<String> testSwAnnotation;
 
-    @TestSwAnnotationVal(onMethod = @SuppressWarnings("unchecked"))
+    @TestSwAnnotationVal(value = @SuppressWarnings("unchecked"))
     private List<String> testSwAnnotationVal;
+
+    @TestSwAnnotationVal(value = {@SuppressWarnings("unchecked")})
+    private List<String> list1;
+
+    @TestSwAnnotationVal({@SuppressWarnings("unchecked"), @SuppressWarnings("unchecked")})
+    private List<String> list2;
 }
 
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.FIELD)
 @interface TestSwAnnotation {
-    SuppressWarnings value();
+    SuppressWarnings value() default @SuppressWarnings({"",});
 }
 
 @Target({ElementType.FIELD, ElementType.TYPE})
 @Retention(RetentionPolicy.SOURCE)
 @interface TestSwAnnotationVal{
 
-    SuppressWarnings[] onMethod() default {};
+    SuppressWarnings[] value() default {@SuppressWarnings({"",})};
 
     @Deprecated
     @Retention(RetentionPolicy.SOURCE)

